### PR TITLE
feat(home-manager/hyprland): source theme + accent

### PR DESCRIPTION
**Pull request by Liassica**
Since Home Manager defbb9c, Hyprland on Home Manager now sources files before most other configuration, meaning ctp-nix can now source a theme colors file instead of inserting them into the main configuration. In addition, accent and accentAlpha color variables have been added in a second sourced file.

### CODE REVIEW

**Comment by Anomalocaridid** [comment]
If files are sourced first, would it be possible to implement an accent option that sets Hyprland variables named something like `$accent` and `$accentAlpha` to the corresponding variables for your accent color?

For example, if your accent is red, would the below lines work if the module added them to your Hyprland config?
```hypr
$accent=$red
$accentAlpha=$redAlpha
```

**Comment by Liassica** [comment]
Added that. Currently it's implemented as a second sourced file, as variables are one of the only configuration that gets placed before `source` entries, so they can't just be added with `hyprland.settings`.

File: modules/home-manager/hyprland.nix [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
-  # the variables appear early enough in the file to be applied properly.
-  config.wayland.windowManager.hyprland.settings = lib.mkIf enable
-    (lib.ctp.fromINI (sources.hyprland + /themes/${cfg.flavour}.conf));
+  config.wayland.windowManager.hyprland.settings.source = [
+    (lib.mkIf enable (sources.hyprland + /themes/${cfg.flavour}.conf))
+    (builtins.toFile "hyprland-${cfg.accent}-accent.conf" ''
+      $accent=''$${cfg.accent}
+      $accentAlpha=''$${cfg.accent}Alpha
+    '')
+  ];
```
Comment (getchoo) [inline modules/home-manager/hyprland.nix:17]:
```suggestion
  config.wayland.windowManager.hyprland.settings = lib.mkIf enable {
    source = [
      "${sources.hyprland}/themes/${cfg.flavour}.conf"
      (builtins.toFile "hyprland-${cfg.accent}-accent.conf" ''
        $accent=''$${cfg.accent}
        $accentAlpha=''$${cfg.accent}Alpha
      '')
    ];
  };
```

**Review by getchoo — CHANGES_REQUESTED** [review]

**Comment by getchoo** [comment]
sorry for the wait! it seems this branch has fallen a bit behind and is still stuck with a bug that's been fixed; could this be rebased?

**Comment by Liassica** [comment]
Done! I also updated the commit description a bit since this is an official Catppuccin repository now.

**Review by getchoo — APPROVED** [review]
so sorry for the long wait. thanks so much!

**Pull request merged**